### ISSUE TITLE
TCP(MySQL) Forwarding not responding

### ISSUE THREAD

**Issue by woshikie** [antoniomika/sish#100]
I'm trying to port forward a local mysql database.
This is my docker-compose.yml for my database instance
```
version: '3'

services:
  db:
    image: mysql:5
    restart: unless-stopped
    environment:
      MYSQL_ROOT_PASSWORD: 'Very Secure Password'
    ports:
      - '33306:3306'
    volumes:
      - ./mount:/var/lib/mysql
```
next, I setup my sish instance on a remote server with 
```
version: "3"
services:
  sish:
    image: antoniomika/sish:latest
    restart: always
    volumes:
      - ./sish/ssl:/ssl
      - ./sish/keys:/keys
      - ./sish/pubkeys:/pubkeys
    network_mode: host
    command: |
      --ssh-address=:22222
      --https-address=:20443
      --http-address=:20080
      --https=false
      --https-certificate-directory=/ssl
      --authentication-keys-directory=/pubkeys
      --private-key-location=/keys/ssh_key
      --domain=<my domain>
      --bind-random-subdomains=false
      --bind-random-ports=false
      --proxy-protocol=false
      --redirect-root=false
```
then, on my local server, i ran:
`ssh -R 3306:localhost:33306 <my domain> -p 22222`
The ssh tunnel service started but when I try to connect to to my server though MySQL Workbench, it timed out.
Tried to tunnel with Ngrok and it works.
It seemed like the no packets are passed through the tunnel

I have other HTTP tunnels running on the sish instance on the remote server too.
HTTP tunnels work very well.

I started using SISH a couple months ago and all the tunnels worked well.
This only happened after I updated to the latest version

EDIT:
Confirmed working with version 0.4.0
<details>
<summary>The docker config:</summary>

```
    volumes:
      - ./sish/ssl:/ssl
      - ./sish/keys:/keys
      - ./sish/pubkeys:/pubkeys
    network_mode: host
    command: |
      -sish.addr=:22222
      -sish.https=:30443
      -sish.http=:30080
      -sish.httpsenabled=false
      -sish.keysdir=/pubkeys
      -sish.pkloc=/keys/ssh_key
      -sish.bindrandom=false
      -sish.domain=<mydomain>
      -sish.forcerandomsubdomain=false
```

</details>

**Comment by antoniomika** [comment]
Thanks for reaching out @woshikie! 

I think this has something to do with the proxy protocol implementation. What seems to be happening is calls to get the underlying `RemoteAddr` are blocked for some reason I haven't been able to find just yet. It's possible this only started showing up after I upgraded the proxyproto package in the v1 upgrade. 

I'll keep digging and hopefully have a fix for you later today or tomorrow. Sorry for the inconvenience.

**Comment by woshikie** [comment]
No problem, take your time. Appreciate your work!

**Comment by antoniomika** [comment]
@woshikie should be good to go with the latest release (v1.0.10). It was indeed the proxy proto lib that I was using. I went ahead and modified the flags with regards to proxy proto and optimized how the header is read so this will work even if proxy proto is used. 

Last thing to note would just be the `--idle-connection` flag as if there is no data being sent between mysql and your client and that flag is set, sish will close the connection because it is technically idle. Mysql has clean reconnect logic though which I also verified works fine.
